### PR TITLE
Adds redis lock

### PR DESCRIPTION
**Pull request by codingjoe**

### CODE REVIEW

File: tests/test_lock.py [hunk 1]
```diff
@@ -66,4 +66,21 @@ def test_cache_lock(self):
         thread1.join()
         thread2.join()
 
+    def test_redis_lock(self):
```
Comment (kmmbvnr) [inline tests/test_lock.py]: 3d party dependent and long-running tests should be optional (ex: test_contrib_celery)

File: tests/urls.py [hunk 2]
```diff
@@ -0,0 +1 @@
+urlpatterns = []
```
Comment (kmmbvnr) [inline tests/urls.py]: This change is not required

File: viewflow/lock.py [hunk 3]
```diff
@@ -1,16 +1,34 @@
...
 
-from django.core.cache import cache
+from django.core.cache import DEFAULT_CACHE_ALIAS
 from django.db import transaction, DatabaseError
+from django.conf import settings
 
 from viewflow.exceptions import FlowLockFailed
 
 
+CACHE_BACKEND = getattr(settings, 'VIEWFLOW_LOCK_CACHE_BACKEND', DEFAULT_CACHE_ALIAS)
```
Comment (kmmbvnr) [inline viewflow/lock.py]: I don't want to introduce any viewflow related settings at this time.

Generally, in order not to get into settings hell, I suppose to design settings as a single `VIEWFLOW` settings dictionary (ex: django TEMPLATES). And all settings access in the viewflow library will goes from `viewflow.settings` module, that will responsible for settings validation and default values.

Comment (kmmbvnr) [inline viewflow/lock.py]: Currently you can just have your own lock impl in your project codebase.

File: viewflow/lock.py [hunk 4]
```diff
@@ -70,10 +88,36 @@ def lock(flow_cls, process_pk):
...
+        finally:
+            cache.delete(key)
 
-        with transaction.atomic():
-            yield
+    return lock
+
+
+def redis_lock(flow, attempts=5, expires=120):
+    """Task lock based redis locks implemented in ``django-redis``."""
```
Comment (kmmbvnr) [inline viewflow/lock.py]: Since this is depends from non-core django, this should be in the 3d party module.

**Comment by kmmbvnr** [comment]
Thanks for the patch. I merged it in efa9c9b8e75fd22187de285bbba98be7b70a7236 and 54af6662a2b94f945270a9c68444779d173b3aa4

File: viewflow/lock.py (hunk 3 above)
Comment (codingjoe) [inline viewflow/lock.py]: I get your point, but why didn't you tell me that before merging. I could have created a callable, that has a configurable cache. Which I will do now.

**Pull request closed without merging**